### DISCUSSION TITLE
A way to split assets and Markdown sources?

### DISCUSSION THREAD

**Discussion by DimanNe** [squidfunk/mkdocs-material#4032] [Q&A] [answered]
I would like to have ONLY markdown files in `docs/` directory (I am not sure why by default we are supposed to mix `.md` files with completely unrelated formatting/style/javascript files).

I was able to achieve this partially:
* `overrides` *are* outside
* `javascripts` *are* outside — I set the following directory in `mkdocs.yml`:
```yml
extra_javascript:
  - ../javascripts/mathjax.js
```

#### But unfortunately there are `stylesheets/` in `docs/`.
Adjusting path in mkdocs.yaml like this:
```yml
extra_css:
  - ../stylesheets/extra.css
```
did not help.

#### Does anyone know how I can extract/move `stylesheets` from `docs/`?

**Comment by facelessuser** [comment]
1. MkDocs Material doesn't control how MkDocs work, I would take such issues up with MkDocs.
2. I don't understand what the complaint is. Whatever you define as your root documentation folder contains all your documentation assets. If you'd like to split them further, you are free to do so: `docs/markdown` and `docs/stylesheets`. Name them however you like and organize them accordingly.

**Reply by DimanNe** [reply]
Unfortunately in this case the site will display "*markdown*" in navigation bar...
(I know that I can provide set navigation hierarchy and names of pages manually...)

**Comment by squidfunk** [comment, marked as answer]
Seconding @facelessuser, this is related to how MkDocs handles assets. However, one more idea: you could move your assets into your overrides. We do the same for our documentation:
https://github.com/squidfunk/mkdocs-material/tree/master/src/overrides

**Reply by DimanNe** [reply]
Thanks for the response!

Is it supposed to "just" work?
I tried moving `docs/stylesheets/extra.css` to `src/assets/stylesheets/extra.css` and commenting the following section in mkdocs.yml:
```
extra_css:
  - stylesheets/extra.css
```
it did not help...

But I noticed in your example that you have much **much** more files in `src/`... one of them refers to stylesheets — https://github.com/squidfunk/mkdocs-material/blob/master/src/base.html#L74 . Do I need to do something similar?

**Reply by squidfunk** [reply]
No, `src` is not the right directory. Note that we're building the theme. I should have made that clearer. It's probably easiest to follow the [theme extension guide](https://squidfunk.github.io/mkdocs-material/customization/#extending-the-theme) and use the following structure:

```
.
└─ overrides/
   └─ assets/
      └─ javascripts/
         └─ extra.js
      └─ stylesheets/
         └─ extra.css
```

Then reference those files in `mkdocs.yml`:

``` yaml
extra_css:
  - assets/stylesheets/extra.css
```

**Reply by DimanNe** [reply]
Wow! It worked!

## Thank you!
